pkg/register/restResolver.go

```diff
@@ -76,8 +76,13 @@ func (c *RestResolverClient) GetDocument(ctx context.Context, documentID string)
 		return nil, err
 	}
 
-	discoverURL := fmt.Sprintf("%s/1.0/discover/%s", c.url.String(), url.QueryEscape(documentID)) // todo: join path?
-	response, err := c.client.Get(discoverURL)
+	discoverURL := fmt.Sprintf("%s/1.0/discover/%s", c.url.String(), url.QueryEscape(documentID))
+
+	reqWithContext, err := http.NewRequestWithContext(ctx, http.MethodGet, discoverURL, nil)
+	if err != nil {
+		return nil, &ResolverError{err: err, errType: ServerError}
+	}
+	response, err := c.client.Do(reqWithContext)
```

Comment:
The below handling won't handle context errors correctly - either
1. Check with the helper (is context error) & just return **or**
2. Add `Is()` (as defined [here](https://pkg.go.dev/errors#Is)) for `ResolverError`

Also, I believe `ReadAll` further down:
1. Can also raise ctx err (since still reading response)
2. Can now use `io.ReadAll` instead from `ioutil` (unless I'm mistaken)